### ISSUE TITLE
Test content created using GC Articles as CMS

### ISSUE THREAD

**Issue by amazingphilippe** [cds-snc/notification-planning#393]
_Test content created using GC Articles as CMS_

# Description

As a Notify team member, I need to be able to make sure the content created in GC Articles is working, bug free and accessible so that I can complete the content migration. 

WHY are we building? Easy to maintain content without interfering in the Notify code. 
WHAT are we building? Some content pages in GC Articles
VALUE created by our solution. Easy to maintain content without interfering in the Notify code. 

# Acceptance Criteria** (Definition of done) 

Given we have drafted content in GC Articles, when we are ready to publish, then we are sure to release a bug free, accessible content update. 

Things to do and figure out before we release the GC Articles collab work:
- [x]  Setting draft pages to published: the styles are okay before pages are switched to published mode (published on staging)
- [x]  A11y tests and scans: the same level of accessibility as the prod website is achieved. [Run these tests](https://trello.com/b/Qj96p840/gc-notify-a11y-assessment)
- [x] GC Articles needs to complete https://github.com/cds-snc/notification-admin/pull/1240 (Caching)
- [ ]  Testing routes: if a slug is equal to an existing route, the existing route is removed in the code (requires help from dev or Phil). 
- [ ] Need to wait for another Notify release for pages to be pushed to production

### QA

- [ ] A11y scan in staging for some EN pages before pushing to prod (Privacy, Security, Why Notify, Features, Guidance)
- [ ] Test links in the footers and in the app to ensure they all go to the right place for both
- [ ] Issues found in testing/bug bash https://docs.google.com/spreadsheets/d/1sWOHBc_gUaFYjkFqkMZQ_PQTl5xwnEztKT1vyUeYVcY/edit#gid=0

**Comment by amazingphilippe** [comment]
Blocked as GC Articles are performing updates to WordPress

**Comment by EliseKa** [comment]
Okay, I reviewed most pages to make sure they looked okay and that the content was up to date, and published the ones that were. Here are the things I want to discuss, as I don't know if another round of front-end changes may be needed:

- [ ] Email formatting guide (EN + FR): seems to me that the output preview displays text too low (not aligned with top of box)
- [ ] Email formatting guide (EN + FR): the boxes act weird when we resize the window in preview mode. They move one above the other but with an offset.
- [ ] All pages: the margins are not the same as in Prod. (Text margins in prod are narrower)
- [ ] All pages: 2 columns in general - I used a 400 px width because that seems to match the text width. A 50% width doesn't match the text width though, it extends on all the screen width
- [ ] Security (EN + FR): the spacing before and after H3 is the same - it should be larger before H3 for better scanning.
- [ ] All pages: it seems to me that the titles font is not the same as before (it's currently different than the font used in prod)
- [ ] Fonctionnalités (FR): images are not at correct position in preview mode. 
- [ ] Features (EN): images are at correct position, but the alignment is too high in preview mode.
- [ ] Why GC Notify (EN + FR): Following the paragraph "Comply with communications requirements", the spacing between paragraphs is very large in preview mode, while there's no additional "Enter" in draft mode.

**Comment by EliseKa** [comment]
New Trello board for a11y testing: https://trello.com/b/HzI1ve9m/gc-articles-notify-public-pages

**Comment by EliseKa** [comment]
Partial route testing update (not all pages are published yet):

**Terms of use**
EN: https://articles.cdssandbox.xyz/notification-gc-notify/terms-of-use/
FR: https://articles.cdssandbox.xyz/notification-gc-notify/fr/conditions-utilisation/
- toggle EN and FR works, but the old FR slug takes precedence (the articles in the slug come back when toggling)
- no problem with old route because current slug is different
- copy-pasting urls works fine with both language

**Privacy**
EN: https://articles.cdssandbox.xyz/notification-gc-notify/privacy/
FR: https://articles.cdssandbox.xyz/notification-gc-notify/fr/confidentialite/
- [ ] need to remove old route in code because it takes precedence
- copy-pasting urls works fine with both languages
- cannot test toggle since old route takes precedence

**Message delivery status**
EN: https://articles.cdssandbox.xyz/notification-gc-notify/message-delivery-status/
FR: https://articles.cdssandbox.xyz/notification-gc-notify/fr/etat-livraison-messages/
- [ ] toggle EN and FR works, but the old FR slug takes precedence (I'm currently looking into this issue)
- no problem with old route because current slug is different
- copy-pasting urls works fine with both languages

**Personalisation guide**
EN: https://articles.cdssandbox.xyz/notification-gc-notify/personalisation-guide/
FR: https://articles.cdssandbox.xyz/notification-gc-notify/fr/guide-personnalisation/
- copy-pasting urls works fine with both languages
- toggle EN and FR works
- no problem with old route because current slug is different

**Accessibility**
EN: https://articles.cdssandbox.xyz/notification-gc-notify/accessibility/
FR: https://articles.cdssandbox.xyz/notification-gc-notify/fr/accessibilite/
- copy-pasting urls works fine with both languages
- toggle EN and FR works
- no problem with old route because current slug is different

**Comment by EliseKa** [comment]
May not be the best, but will serve our purpose for now - [Spreadsheet for listing a11y issues](https://docs.google.com/spreadsheets/d/1KB-RXe30azptukvSD6OME7XFdsdG3YTRTCgM1UuWAEM/edit#gid=0)

**Comment by amazingphilippe** [comment]
CSS fixes for this story: https://github.com/cds-snc/notification-admin/pull/1232

Ready to review

**Comment by adriannelee** [comment]
Élise continue this work today - updates in the linked links

**Comment by EliseKa** [comment]
New style changes required following PR #1232:

[ ] Why Notify page: the issue of the icon being too high is resolved, however the spacing between the icon and the text is very large.
![Capture d’écran, le 2022-03-02 à 10.32.17.png](https://images.zenhubusercontent.com/6140b56b70c730353f1cd567/7d646fa5-454a-43de-9881-d0c713642205) 

[ ] All pages: The bullet points seem to have disappeared
![Capture d’écran, le 2022-03-02 à 10.35.47.png](https://images.zenhubusercontent.com/6140b56b70c730353f1cd567/4730240f-70ae-4e76-b9f5-b2e063503b7d)

**Comment by amazingphilippe** [comment]
New fixes in https://github.com/cds-snc/notification-admin/pull/1236

**Comment by EliseKa** [comment]
@amazingphilippe Just want to check with you if you think we need other fixes or not:
- The bullet points are not indented as they used to be (they're left aligned). I think it helps scanning when they're indented, but not sure.
![Capture d’écran, le 2022-03-03 à 14.09.52.png](https://images.zenhubusercontent.com/6140b56b70c730353f1cd567/f7f913e1-44bd-46a8-8da3-d39f525c63d0)


- The line spacing in the tables changes when resizing the viewport, and there's still a lot of white space (but it's better than before).
Normal viewport:
![Capture d’écran, le 2022-03-03 à 14.12.46.png](https://images.zenhubusercontent.com/6140b56b70c730353f1cd567/138e4b0e-517c-4ea3-a4a5-8fba9b140239)

Narrow viewport: 
![Capture d’écran, le 2022-03-03 à 14.16.18.png](https://images.zenhubusercontent.com/6140b56b70c730353f1cd567/195903de-fc3b-4980-a651-ac2d6203c65b)

**Comment by EliseKa** [comment]
Other new fixes in https://github.com/cds-snc/notification-admin/pull/1237, not merged yet.

**Comment by yaelberger-commits** [comment]
Elise working on a11y testing today, needs dev help on one page from GC Articles team, have to add possibility to add permalink for bilingualism

**Comment by EliseKa** [comment]
Update on home page url: creating a FR home page will be on the Notify routing side. [Link to new issue for this](https://app.zenhub.com/workspaces/notify-planning-614b3ad91bc2030015ed22f5/issues/cds-snc/notification-planning/460)

**Comment by EliseKa** [comment]
Not working on this card today so moving back into sprint backlog.

**Comment by EliseKa** [comment]
**On removing old routes**

Had a chat with Jimmy - it may be a task more involved than it seems and the devs should look at all potentialities because urls kind of touch to the "nervous system" of Notify. We agreed that I'll create a PR to get an instance running and allow for testing, and I'll also create a new card specifically for that task.

**Comment by EliseKa** [comment]
Marking this card as currently blocked since we need dev work to make the routes work. 
Link to Zenhub card for that work: [Remove existing routes using same url as routes in GC Articles](https://app.zenhub.com/workspaces/notify-planning-614b3ad91bc2030015ed22f5/issues/cds-snc/notification-planning/500)

Link to PR to set off this work: [Remove routes that override GC Articles routes #1248](https://github.com/cds-snc/notification-admin/pull/1248)

**Comment by EliseKa** [comment]
Also marking the card blocked by other dev work that needs to be done to make the home page bilingual.

**Comment by yaelberger-commits** [comment]
Decision to scope this work down to keeping URLs the same 
Set up GC articles to have the same URLS we currently have for now
No bilingual routes
Stop further work on Google Analytics and Google tag Manager for now, pick up those cards later

**Comment by yaelberger-commits** [comment]
Wordpress doesn't accept unilingual slugs, Phil looking into this today

**Comment by amazingphilippe** [comment]
The PR introduces two ways in which in can finish this task. Looking for some code review, its very repetitive and I think we can make it less bad code.

**Comment by yaelberger-commits** [comment]
Phil is doing 2 things: 
- Redirecting pages to keep the old routes
- ?
Needs a reviewer to optimize code

**Comment by yaelberger-commits** [comment]
Not losing any formatting on pages
a11y scans on published links 
Can test again ally scan in staging before pushing to prod

**Comment by yaelberger-commits** [comment]
@yaelberger-commits Will get a status update from GC Articles team on caching PR and then hand back to Phil once that blocker has been removed

**Comment by yaelberger-commits** [comment]
Jimmy and Andrew reviewed GC Articles caching PR, made some comments, team making some changes, Jimmy and Andrew to hopefully take another pass today.

**Comment by jimleroyer** [comment]
The GCArticles caching feature was merged and integrated into the mainline. That should not be a blocker anymore. Phil and Dave are meeting to sync on latest changes and know what's the best way forward on this task.

**Comment by amazingphilippe** [comment]
#1240 is merged. Working with Dave to update the existing routes.

**Comment by yaelberger-commits** [comment]
Review has been done, but want to reassess a change. Needs a few hours of work left

**Comment by yaelberger-commits** [comment]
Hierarchy of templates was changed with GC Articles. Lost a few elements along the way.

**Comment by yaelberger-commits** [comment]
Functionally complete, Elise will test that content is coming through ok

**Comment by yaelberger-commits** [comment]
Checks from @EliseKa to see if this is good to go

**Comment by andrewleith** [comment]
During our test hack session, the following issues were found:
- [ ] relative links used on some pages (why-notify, features)
- [ ] Links in formatting guide should not leave site
- [ ] "Why notify" menu item not active when on page
- [ ] Image on `/message-delivery-status` pointing to staging
- [ ] "Your account" menu going underneath blue hero banner on home page
- [ ] Mobile language switch not working
- [ ] Incorrect links on `/security` page
